docs/language/blocks-and-lambdas.mdx

```diff
@@ -459,24 +501,89 @@ Syntactically, it looks a bit cryptic when blocks are stored in local
 variables. However, the main purpose of blocks is to pass snippets of code to
 functions.
 
-Blocks aren't just "normal" functions as in other programming languages. They
-are much more efficient, but also come with some restrictions.
-
-### Restrictions
+#### Restrictions
 
 Blocks are stack allocated, which is what makes them so efficient, but it also limits where they can be used:
 
 - Blocks cannot be stored in instance fields, static class fields, and globals.
-
 - Blocks cannot be stored in any collection.
-
 - Blocks cannot be returned from functions and methods.
+- Lambdas cannot capture blocks.
+- Blocks cannot take blocks as arguments.
 
-- [Lambdas](../tasks) cannot capture blocks.
+## Lambdas in Detail
 
-- Blocks cannot take blocks as arguments.
+### Basic Use of Lambdas
+
+Lambdas are more versatile than blocks and can be used in a wider variety of
+situations, though at a cost of reduced efficiency compared to blocks:
+
+```toit
+start-task:
+  task::
+    3.repeat:
+      print "still running"
+      sleep --ms=300
+
+main:
+  start-task
+  print "task started"
+```
+
+In this example, we define a lambda that runs as a separate task, continuously
+printing a message. While less efficient than a block, a lambda is necessary here
+due to its ability to be stored and executed later.
+
+### Storing and Reusing Lambdas
+
+Unlike blocks, lambdas can be stored in variables, fields, and collections:
 
-## Defining functions that take block arguments
+```toit
+class WatchedBox:
+  value_/any := null
+  callbacks_/List ::= []
+
+  watch callback/Lambda:
+    callbacks_.add callback
+
+  value -> any: return value_
+  value= new-value:
+    value_ = new-value
+    callbacks_.do: it.call new-value
+
+add-printer box/WatchedBox:
+  box.watch:: print "Was changed to $it."
```

Comment:
```suggestion
  box.watch:: print "was changed to $it."
```

Reply:
I think I would drop the trailing period here (and in the comments). It just seems misplaced and we don't typically have that in debug prints.

Reply:
done.